projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -215,4 +215,22 @@ public function add_theme_install_menu( $wp_admin = false ) {
 
 		add_submenu_page( $parent_menu_slug, esc_attr__( 'Add New Theme', 'jetpack' ), __( 'Add New Theme', 'jetpack' ), 'install_themes', 'theme-install.php', null, 1 );
 	}
+
+	/**
+	 * Adds Users menu.
+	 *
+	 * @param bool $wp_admin Optional. Whether links should point to Calypso or wp-admin. Default false (Calypso).
+	 */
+	public function add_users_menu( $wp_admin = false ) {
+		parent::add_users_menu( $wp_admin );
+
+		// No need to add a menu linking to WP Admin if there is already one.
+		if ( $wp_admin ) {
```

Comment:
I will let @obenland to confirm, but I think this fall into what he said in paYJgx-1lp-p2#comment-1660:

> > The idea is that instead of defaulting to WP Admin, we default to Calypso which has some features not available to WP Admin.
> Let’s not add any escape hatches to any Calypso screens. This is what we built the toggle for, to change between those two screens for users who feel strongly about it.

Reply:
That's another close one. I think I'm leaning towards not keeping Calypso links. If I understand tt correctly, Atomic users can already add users to their site that are not part of WP.com

Reply:
Through `[domain]/wp-admin/users.php`, they can create a non WP.com user that cannot login cause `[domain]/wp-admin` and `[domain]/wp-login` redirects to `https://wordpress.com/log-in` where the user has not been created.

Reply:
> Through `[domain]/wp-admin/users.php`, they can create a non WP.com user that cannot login cause `[domain]/wp-admin` and `[domain]/wp-login` redirects to `https://wordpress.com/log-in` where the user has not been created.

Except when SSO is disabled. In such cases they can use the `[domain]/wp-login.php` URL for logging in, as advised in the email received when the user is created:

<img width="943" alt="Screen Shot 2021-02-17 at 17 38 59" src="https://user-images.githubusercontent.com/1233880/108236991-c401c880-7147-11eb-9e26-d7ca5c0573d3.png">

I think we should consider the current status of the SSO module and adjust the menus as follows:
- If SSO is enabled, always link to Calypso, regardless the link destination setting toggle. There is no point on linking to `/wp-admin/users.php`, since the extra features available there don't work when SSO is enabled.
- If SSO is disabled and the toggle is enabled, always link to WP Admin in order to respect the toggle choice. Do not include any link to Calypso, since the ability to invite a user is quite similar to create a new user.
- If SSO is disabled and the toggle is enabled, always link to Calypso in order to respect the toggle choice. Do not include any link to WP Admin, since the ability to create a user is quite similar to invite a new user.

Reply:
Good idea @mmtr! 

> If SSO is enabled, always link to Calypso, regardless the link destination setting toggle. There is no point on linking to /wp-admin/users.php, since the extra features available there don't work when SSO is enabled.

I am afraid that `/wp-admin/users.php` and specifically the edit user screen has features (extra fields eg Billing Address for Woo) that Calypso doesn't have.

Reply:
> extra fields eg Billing Address for Woo

I believe these are intended for customers and not for site administrators, so likely can be managed in WooCommerce > Customers.

Reply:
Let's go for it and we can iterate if needed!

Reply:
After thinking more deeply about this (https://github.com/Automattic/wp-calypso/issues/49211#issuecomment-782120308), I think the current proposed solution can be enough. Let's revisit it if needed after publishing the 2nd beta.